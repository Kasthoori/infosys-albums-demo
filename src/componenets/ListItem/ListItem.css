.ListItem {
  position: relative;
  width: 100%;
  height: 50px;
  background-color: #e0e0e0;
  border-radius: 10px;
  margin-bottom: 4px;
  display: flex;
  flex-direction: row;
  align-items: center;
  padding-left: 20px;
}

.ListItem:hover {
  background-color: #bdbdbd;
  transition: 0.3s;
}

#number {
  font-size: 20px;
  font-weight: bold;
}

#txt {
  position: absolute;
  left: 60px;
}
