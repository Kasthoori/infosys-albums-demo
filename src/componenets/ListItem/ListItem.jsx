import React from "react";
import "./ListItem.css";

const ListItem = (props) => {
  const handleClick = props.onClick;
  const id = props.id;
  const title = props.title;
  const userId = props.userId;

  return (
    <div className="ListItem" onClick={() => handleClick(id, title, userId)}>
      <span id="number">{props.id}</span>
      <br />
      <span id="txt">{props.title}</span>
    </div>
  );
};

export default ListItem;
