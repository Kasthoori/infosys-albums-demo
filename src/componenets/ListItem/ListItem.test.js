import React from "react";
import { shallow } from "enzyme";
import ListItem from "./ListItem";
import toJson from "enzyme-to-json";

const listOnClick = jest.fn();

describe("ListItem", () => {
  let wrapper;
  const item = {};

  beforeEach(() => (wrapper = shallow(<ListItem />)));

  it("should render <div />", () => {
    expect(wrapper.find("div").length).toEqual(1);
  });

  it("ListItems renders witshout crashing", () => {
    const cont = shallow(<ListItem />);
    expect(toJson(cont)).toMatchSnapshot();
  });
});
