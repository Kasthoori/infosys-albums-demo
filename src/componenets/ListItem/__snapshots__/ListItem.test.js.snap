// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem ListItems renders witshout crashing 1`] = `
<div
  className="ListItem"
  onClick={[Function]}
>
  <span
    id="number"
  />
  <br />
  <span
    id="txt"
  />
</div>
`;
